app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -218,6 +223,8 @@ public final class Player implements PlaybackListener, Listener {
     private boolean isPrepared = false;
     private boolean wasPlaying = false;
 
+    private boolean defaultResolutionSetForManifestsSources = false;
```

Comment:
What is this variable? Why are you setting it to `false` only when Next and Previous are called? Since `setDefaultResolutionForManifestSourcesIfNeeded` does not depend on anything, shouldn't it be called only when `trackSelector` is initialized, i.e. in the `Player` constructor?